packages/core/src/components/steps-modal/steps-modal.spec.ts

```diff
@@ -0,0 +1,154 @@
+            >
+        </atom-steps-modal>
+    `)
+  })
+  it('should change step when handlePrimary is called', async () => {
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+    await page.waitForChanges()
+
+    expect(
+      page.root?.querySelector('atom-modal')?.getAttribute('header-title')
+    ).toBe('Step 2')
+
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+
+    await page.waitForChanges()
+
+    expect(
+      page.root?.querySelector('atom-modal')?.getAttribute('header-title')
+    ).toBe('Step 3')
+  })
+
+  it('should change step when handleSecondary is called', async () => {
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+    await page.waitForChanges()
+
+    expect(
+      page.root?.querySelector('atom-modal')?.getAttribute('header-title')
+    ).toBe('Step 2')
+
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomSecondaryClick'))
+
+    await page.waitForChanges()
+
+    expect(
+      page.root?.querySelector('atom-modal')?.getAttribute('header-title')
+    ).toBe('Step 1')
+  })
+
+  it('should emit atomFinish when last step is reached', async () => {
+    const finishSpy = jest.fn()
+
+    page.root?.addEventListener('atomFinish', finishSpy)
+
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+    await page.waitForChanges()
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+    await page.waitForChanges()
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+    await page.waitForChanges()
+
+    expect(finishSpy).toHaveBeenCalled()
+  })
+
+  it('should not change step when handleSecondary is called on first step', async () => {
+    const cancelSpy = jest.fn()
+
+    page.root?.addEventListener('atomCancel', cancelSpy)
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomSecondaryClick'))
+    await page.waitForChanges()
+
+    expect(
+      page.root?.querySelector('atom-modal')?.getAttribute('header-title')
+    ).toBe('Step 1')
+    expect(cancelSpy).toHaveBeenCalled()
+  })
+  it('should apply show class to current step and hide to others', async () => {
+    let stepShow = page.root?.querySelector('.show')
+
+    expect(stepShow).not.toBeNull()
+
+    expect(stepShow?.querySelector('div')?.textContent).toBe('Step 1 Content')
+
+    page.root
+      ?.querySelector('atom-modal')
+      ?.dispatchEvent(new CustomEvent('atomPrimaryClick'))
+
+    await page.waitForChanges()
+
+    stepShow = page.root?.querySelector('.show')
+
+    expect(stepShow).not.toBeNull()
+
+    expect(stepShow?.querySelector('div')?.textContent).toBe('Step 2 Content')
+  })
+})
```

Comment:
congrats, man! amazing tests. I'm missing one to coverage when `currentStep` is provided

Reply:
thanks!! I create one for this case here [693539c](https://github.com/juntossomosmais/atomium/pull/599/commits/693539c1609736fc4c7eaffbc565d0631bc03425)